src/pageServices/contentScripts/highlight.js

```diff
@@ -242,7 +242,7 @@ export const highlightOnPage = () => {
 
   const findAndHighlight = (param) => {
     if (param) {
-      if (!predictedElements) predictedElements = param.elements;
+      if (!predictedElements || predictedElements.length != param.elements.length) predictedElements = param.elements;
```

Comment:
hope you'll be able to add new element with `sendMessage.addElement`, it will allow you to avoid changing this logic

Reply:
sorry, I can't get the logic. Which case is handled here?

Reply:
I refactored some parts of adding custom locator and return this peace as it was
[[d679840](https://github.com/jdi-testing/jdn-ai/pull/1169/commits/d679840bcd1305c969f56d6fd35183ed0b10b623)](https://github.com/jdi-testing/jdn-ai/pull/1169/commits/d679840bcd1305c969f56d6fd35183ed0b10b623)